rados/conn.go

```diff
@@ -71,9 +71,6 @@ func (c *Conn) Connect() error {
 
 // Shutdown disconnects from the cluster.
 func (c *Conn) Shutdown() {
-	if err := c.ensureConnected(); err != nil {
-		return
-	}
```

Comment:
What's this change have to do with the rest of the patch?

Reply:
```
// freeConn releases resources that are allocated while configuring the
// connection to the cluster. rados_shutdown() should only be needed after a
// successful call to rados_connect(), however if the connection has been
// configured with non-default parameters, some of the parameters may be
// allocated before connecting. rados_shutdown() will free the allocated
// resources, even if there has not been a connection yet.
```

Doing the check will result in a memory leak if there is no connection based on what it says in the src documentation.